pkg/scheduler/localscheduler.go

```diff
@@ -0,0 +1,167 @@
+package scheduler
+
+import (
+	"context"
+	"encoding/json"
+	"fmt"
+
+	"github.com/google/uuid"
+	"github.com/kyma-incubator/reconciler/pkg/cluster"
+	"github.com/kyma-incubator/reconciler/pkg/keb"
+	"github.com/kyma-incubator/reconciler/pkg/model"
+	"go.uber.org/zap"
+	"golang.org/x/sync/errgroup"
+)
+
+type LocalSchedulerOption func(*LocalScheduler)
+
+func WithLogger(logger *zap.SugaredLogger) LocalSchedulerOption {
+	return func(ls *LocalScheduler) {
+		ls.logger = logger
+	}
+}
+
+func WithComponentsWithCRDs(components []string) LocalSchedulerOption {
+	return func(ls *LocalScheduler) {
+		ls.componentsWithCRDs = components
+	}
+}
+
+func WithPrerequisites(components []string) LocalSchedulerOption {
+	return func(ls *LocalScheduler) {
+		ls.prerequisites = components
+	}
+}
+
+type LocalScheduler struct {
+	workerFactory      WorkerFactory
+	logger             *zap.SugaredLogger
+	componentsWithCRDs []string
+	prerequisites      []string
+}
+
+func NewLocalScheduler(workerFactory WorkerFactory, opts ...LocalSchedulerOption) *LocalScheduler {
+	ls := &LocalScheduler{
+		workerFactory: workerFactory,
+		logger:        zap.NewNop().Sugar(),
+	}
+
+	for _, opt := range opts {
+		opt(ls)
+	}
+
+	return ls
+}
+
+func (ls *LocalScheduler) Run(ctx context.Context, c keb.Cluster) error {
+	schedulingID := uuid.NewString()
+
+	clusterState, err := localClusterState(&c)
+	if err != nil {
+		return fmt.Errorf("failed to convert to cluster state: %s", err)
+	}
+
+	components, err := clusterState.Configuration.GetComponents()
+	if err != nil {
+		return fmt.Errorf("failed to get components: %s", err)
+	}
+
+	for _, c := range components {
+		if contains(ls.prerequisites, c.Component) {
```

Comment:
Suggestion: Reduce nesting by continuing early:
```suggestion
		if !contains(ls.prerequisites, c.Component) {
		     continue
		}
```

Reply:
Done